plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,471 @@
+}
+
+func (plugin *Fritzbox) Init() error {
+	if plugin.Log == nil {
+		plugin.Log = logger.New("inputs", pluginName, "")
+	}
+	err := plugin.initDeviceClients()
+	if err != nil {
+		return err
+	}
+	plugin.initServiceHandlers()
+	return nil
+}
+
+func (plugin *Fritzbox) initDeviceClients() error {
+	for _, device := range plugin.Devices {
+		deviceUrl, err := url.Parse(device)
+		if err != nil {
+			return err
+		}
+		client := tr064.NewClient(deviceUrl)
+		client.Debug = plugin.Debug
+		client.Timeout = time.Duration(plugin.Timeout)
+		client.InsecureSkipVerify = plugin.TlsSkipVerify
+		plugin.deviceClients = append(plugin.deviceClients, client)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) initServiceHandlers() {
+	if plugin.DeviceInfo {
+		plugin.serviceHandlers[deviceinfo.ServiceShortType] = plugin.gatherDeviceInfo
+	}
+	if plugin.WanInfo {
+		plugin.serviceHandlers[wancommonifconfig.ServiceShortType] = plugin.gatherWanInfo
+	}
+	if plugin.PppInfo {
+		plugin.serviceHandlers[wanpppconn.ServiceShortType] = plugin.gatherPppInfo
+	}
+	if plugin.DslInfo {
+		plugin.serviceHandlers[wandslifconfig.ServiceShortType] = plugin.gatherDslInfo
+	}
+	if plugin.WlanInfo {
+		plugin.serviceHandlers[wlanconfig.ServiceShortType] = plugin.gatherWlanInfo
+	}
+	if plugin.HostsInfo {
+		plugin.serviceHandlers[hosts.ServiceShortType] = plugin.gatherHostsInfo
+	}
+}
+
+func (plugin *Fritzbox) Gather(acc telegraf.Accumulator) error {
+	start := time.Now()
+	for _, deviceClient := range plugin.deviceClients {
+		plugin.gatherDevice(acc, deviceClient)
+	}
+	if plugin.Debug {
+		elapsed := time.Since(start)
+		plugin.Log.Info("took: ", elapsed)
+	}
+	plugin.queryCycleCounter++
+	if plugin.queryCycleCounter >= plugin.FullQueryCycle {
+		plugin.queryCycleCounter = 0
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherDevice(acc telegraf.Accumulator, deviceClient *tr064.Client) {
+	if plugin.Debug {
+		plugin.Log.Infof("Querying %s", deviceClient.DeviceUrl.Redacted())
+	}
+	services, err := deviceClient.Services(tr064.DefaultServiceSpec)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	for _, service := range services {
+		serviceHandler := plugin.serviceHandlers[service.ShortType()]
+		if serviceHandler == nil {
+			continue
+		}
+		err := serviceHandler(acc, deviceClient, service)
+		if err != nil {
+			acc.AddError(err)
+			return
+		}
+	}
+}
+
+type serviceHandlerFunc func(telegraf.Accumulator, *tr064.Client, tr064.ServiceDescriptor) error
+
+func (plugin *Fritzbox) gatherDeviceInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := deviceinfo.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &deviceinfo.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["fritz_device"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["fritz_service"] = serviceClient.Service.ShortId()
```

Comment:
```suggestion
	tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
	tags["service"] = serviceClient.Service.ShortId()
```
`source` to better align with other plugins, also I think the `fritz_` prefix can be left out since it is already a `fritzbox_` metric?

Reply:
Will rename

Reply:
Done via 956766ca069329ce108d2513ab3f4d970f249921